app/src/main/java/com/stevesoltys/seedvault/restore/install/ApkSplitCompatibilityChecker.kt

```diff
@@ -0,0 +1,118 @@
+package com.stevesoltys.seedvault.restore.install
+
+import android.content.Context
+import android.os.Build
+import android.util.Log
+
+private const val TAG = "SplitCompatChecker"
+
+private const val CONFIG_PREFIX = "config."
+private const val CONFIG_LENGTH = CONFIG_PREFIX.length
+
+private const val LDPI_VALUE = 120
+private const val MDPI_VALUE = 160
+private const val TVDPI_VALUE = 213
+private const val HDPI_VALUE = 240
+private const val XHDPI_VALUE = 320
+private const val XXHDPI_VALUE = 480
+private const val XXXHDPI_VALUE = 640
+
+class DeviceInfo(context: Context) {
+    val densityDpi: Int = context.resources.displayMetrics.densityDpi
+    val supportedABIs: List<String> = Build.SUPPORTED_ABIS.toList()
+}
+
+/**
+ * Tries to determine APK split compatibility with a device by examining the list of split names.
+ * This only looks on the supported ABIs and the screen density.
+ * Other config splits e.g. based on OpenGL or Vulkan version are also possible,
+ * but don't seem to be widely used, so we don't consider those for now.
```

Comment:
I feel like we've only gone through a small subset of apps, I don't remember seeing any newer games on the list which would be most likely to utilize such a split.

Reply:
I checked several games on apk mirror, but most just used arch based splits, some also languages. Please try to find more. I agree that it would be best to have a much larger data basis for this.